### PR TITLE
11 get api articles article id comment count

### PR DESCRIPTION
**Pull request by AlexandraB123**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -70,6 +70,93 @@ describe("/api/articles", () => {
...
+            });
+          });
+        });
+    });
+    test("GET: 200. Returns empty array if given valid search topic when no articles are on that topic", () => {
+      return request(app)
+        .get("/api/articles?topic=paper")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toEqual([]);
```
Comment (xViolaine) [inline __tests__/api.test.js]: good job thinking of this test!

File: __tests__/api.test.js [hunk 2]
```diff
@@ -70,6 +70,93 @@ describe("/api/articles", () => {
...
+      return request(app)
+        .get("/api/articles?order=not_asc_or_desc")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toBeSorted({
+            key: "created_at",
+            descending: true,
+          });
+        });
+    });
```
Comment (xViolaine) [inline __tests__/api.test.js]: In general, I like the idea of doing it like this, but the user definitely should see an error/be informed there is an error when they are putting in orders or topics etc. that are incorrect.

I'd argue due to that, that we'd want to see a lot more 400/404 errors here in the testing and that the error handling might need some changing!

**Review by xViolaine — CHANGES_REQUESTED** [review]
The way that non existent/wrong queries were handled should be changed!

File: __tests__/api.test.js (hunk 2 above)
Comment (AlexandraB123) [reply __tests__/api.test.js]: Noted, thanks. Model and tests refactored to return 400/404 errors.

File: __tests__/api.test.js [hunk 3]
```diff
@@ -87,6 +192,7 @@ describe("/api/articles/:article_id", () => {
             topic: expect.any(String),
             created_at: expect.any(String),
             votes: expect.any(Number),
+            comment_count: 11,
```
Comment (mezzDavies) [inline __tests__/api.test.js:195]: Really we need to make a new test for this - to be following the principals of TDD. I know it feels a bit strange but please revert this test and add a new one specifically checking for the comment_count property.

File: models/articles.models.js [hunk 4]
```diff
@@ -1,24 +1,58 @@
...
-    SELECT users.username AS author, articles.title, articles.article_id, articles.topic, articles.body, articles.created_at, articles.votes
-    FROM articles
-    LEFT JOIN users ON articles.author = users.username
-    WHERE article_id = $1;
-  `;
+  SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.body, articles.created_at, articles.votes, CAST(COUNT(comments.comment_id) AS INT) AS comment_count
+  FROM articles
+  LEFT JOIN comments ON articles.article_id = comments.article_id
+  WHERE articles.article_id = $1
+  GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.body, articles.created_at, articles.votes;
```
Comment (mezzDavies) [inline models/articles.models.js:51]: We actually only need to group by article.article_id here as that is the unique identifier.

**Review by mezzDavies — APPROVED** [review]
Although I left a couple of comments this is approved for a merge as you have definitely done the bulk of what was required - all good. As ever please do note the comments and make those changes before merging 🙏 . Cheers

File: models/articles.models.js (hunk 4 above)
Comment (AlexandraB123) [reply models/articles.models.js:51]: Done

File: __tests__/api.test.js (hunk 3 above)
Comment (AlexandraB123) [reply __tests__/api.test.js:195]: Done

**Pull request merged**